classes/author/Collector.inc.php

```diff
@@ -1,54 +1,99 @@
 <?php
 /**
- * @file classes/services/QueryBuilders/PKPAuthorQueryBuilder.php
+ * @file classes/author/Collector.inc.php
  *
  * Copyright (c) 2014-2021 Simon Fraser University
  * Copyright (c) 2000-2021 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
- * @class PKPAuthorQueryBuilder
- * @ingroup query_builders
+ * @class author
  *
- * @brief Class for building database queries for authors
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
  */
 
-namespace PKP\services\queryBuilders;
+namespace PKP\author;
 
+use Illuminate\Database\Query\Builder;
 use Illuminate\Support\Facades\DB;
-
+use PKP\core\interfaces\CollectorInterface;
 use PKP\plugins\HookRegistry;
-use PKP\services\queryBuilders\interfaces\EntityQueryBuilderInterface;
 
-class PKPAuthorQueryBuilder implements EntityQueryBuilderInterface
+class Collector implements CollectorInterface
 {
-    /** @var array get authors for one or more contexts */
-    protected $contextIds = [];
+    public const ORDERBY_SEQUENCE = 'a.seq';
+    public const ORDERBY_ID = 'a.author_id';
```

Comment:
Let's make these unique strings rather than table column references. For example, `public const ORDERBY_SEQUENCE = 'sequence';`. That way, if we ever change the table structure, we don't need to change the constant values (which might be used, for example, by a plugin).

Reply:
:+1: for this, but for another reason -- in my case it wasn't sufficient to just list sort by columns, the getQueryBuilder code needed to do some logic based on the selected option (and I didn't want to `switch` by a list of column names).

Reply:
Changed to use something like [this](https://github.com/pkp/pkp-lib/blob/main/classes/submission/Collector.inc.php#L212).